pkg/detectors/jdbc/mysql.go

```diff
@@ -50,47 +51,54 @@ func isMySQLErrorDeterminate(err error) bool {
 	return false
 }
 
-func parseConnStr(connStr string) (hostAndDB, params string, err error) {
+func parseMySQL(subname string) (jdbc, error) {
 	// expected form: [subprotocol:]//[user:password@]HOST[/DB][?key=val[&key=val]]
-	hostAndDB, params, found := strings.Cut(connStr, "?")
-	if !strings.HasPrefix(hostAndDB, "//") {
-		return "", "", errors.New("expected host to start with //")
-	}
-	if !found {
-		return hostAndDB, "", nil
-	}
-	splitParams := strings.Split(params, "&")
-	for i, param := range splitParams {
-		if strings.Contains(strings.ToLower(param), "allowallfiles") {
-			splitParams[i] = "allowAllFiles=false"
-		}
+	if !strings.HasPrefix(subname, "//") {
+		return nil, errors.New("expected host to start with //")
 	}
-	params = strings.Join(splitParams, "&")
 
-	return hostAndDB, params, nil
-}
+	// need for hostnames that have tcp(host:port) format required by this database driver
+	cfg, err := mysql.ParseDSN(strings.TrimPrefix(subname, "//"))
+	if err == nil {
+		return &mysqlJDBC{
+			conn:     subname[2:],
+			userPass: cfg.User + ":" + cfg.Passwd,
+			host:     fmt.Sprintf("tcp(%s)", cfg.Addr),
+			database: cfg.DBName,
+			params:   "timeout=5",
+		}, nil
+	}
 
-func parseMySQL(subname string) (jdbc, error) {
-	hostAndDB, params, err := parseConnStr(subname)
+	// for standard URI format, which is all i've seen for JDBC
+	u, err := url.Parse(subname)
 	if err != nil {
 		return nil, err
 	}
 
-	userPassAndHostAndDB := strings.TrimPrefix(hostAndDB, "//")
-	userPass, hostAndDB, found := strings.Cut(userPassAndHostAndDB, "@")
-	if !found {
-		hostAndDB = userPass
-		userPass = ""
+	dbName := strings.TrimPrefix(u.Path, "/")
+
+	user := ""
+	pass := ""
+	if u.User != nil {
+		user = u.User.Username()
+		pass, _ = u.User.Password()
 	}
-	host, database, found := strings.Cut(hostAndDB, "/")
-	if !found {
-		return nil, errors.New("expected host and database to be separated by /")
+
+	for k, v := range u.Query() {
```

Comment:
What if user/pass aren't included in `u.User` or `q.Query`?

Reply:
I've updated this to default to `root`, and to only provide the username if there is no password.